screwdriver/api/api.go

```diff
@@ -33,102 +38,212 @@ type ResponseError struct {
-	} `json:"habitat"`
-	Docker struct {
-		Image   string `json:"image"`
-		Command string `json:"command"`
-	} `json:"docker"`
-	Binary struct {
-		File string `json:"file"`
-	} `json:"binary"`
-	PipelineId int `json:"pipelineId"`
-}
-
 func (e ResponseError) Error() string {
 	return fmt.Sprintf("Screwdriver API %d %s: %s", e.StatusCode, e.Reason, e.Message)
 }
 
 // New returns API object
-func New(sdAPI, sdToken string) (API, error) {
-	c, err := newClient(sdAPI, sdToken)
-	if err != nil {
-		return nil, err
-	}
-	return API(c), nil
+func New(sdAPI, sdToken string) API {
+	c := newClient(sdAPI, sdToken)
+	return API(c)
 }
 
-func newClient(sdAPI, sdToken string) (*client, error) {
-	c := &client{
+func newClient(sdAPI, sdToken string) *client {
+	return &client{
 		baseURL: sdAPI,
 		jwt:     sdToken,
 		client:  &http.Client{Timeout: timeoutSec * time.Second},
 	}
-	return c, nil
 }
 
-func handleResponse(res *http.Response) ([]byte, error) {
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		return nil, fmt.Errorf("Reading response Body from Screwdriver API: %v", err)
-	}
-
-	switch res.StatusCode / 100 {
+func handleResponse(bodyBytes []byte, statusCode int) ([]byte, error) {
+	switch statusCode / 100 {
 	case 2:
-		return body, nil
+		res := new(util.CommandSpec)
+		err := json.Unmarshal(bodyBytes, res)
+		if err != nil {
+			return nil, fmt.Errorf("Screwdriver API Response unparseable: status=%d, err=%v", statusCode, err)
+		}
+		return bodyBytes, nil
 	case 4:
 		res := new(ResponseError)
-		err = json.Unmarshal(body, res)
+		err := json.Unmarshal(bodyBytes, res)
 		if err != nil {
-			return nil, fmt.Errorf("Unparseable error response from Screwdriver API: %v", err)
+			return nil, fmt.Errorf("Screwdriver API Response unparseable: status=%d, err=%v", statusCode, err)
 		}
 		return nil, res
 	case 5:
-		return nil, fmt.Errorf("%v: Screwdriver API has internal server error", res.StatusCode)
+		return nil, fmt.Errorf("Screwdriver API has internal server error: statusCode=%d", statusCode)
 	default:
-		return nil, fmt.Errorf("Unknown error happen while communicate with Screwdriver API")
+		return nil, fmt.Errorf("Unknown error happen while communicate with Screwdriver API: Statuscode=%d", statusCode)
 	}
 }
 
 // GetCommand returns Command from Screwdriver API
-func (c client) GetCommand(smallSpec *Command) (*Command, error) {
-	cmd := new(Command)
+func (c client) GetCommand(smallSpec *util.CommandSpec) (*util.CommandSpec, error) {
 	uri, err := url.Parse(c.baseURL)
 	if err != nil {
-		return nil, fmt.Errorf("The base Screwdriver API is invalid %q", c.baseURL)
+		return nil, fmt.Errorf("Failed to parse URL on GET: %v", err)
 	}
 	uri.Path = path.Join(uri.Path, "commands", smallSpec.Namespace, smallSpec.Name, smallSpec.Version)
-	req, err := http.NewRequest("GET", uri.String(), strings.NewReader(""))
+
+	// Request to api
+	const contentType = "application/json"
+	// No payload
+	payload := bytes.NewBuffer([]byte(""))
```

Comment:
I think just declaring a payload variable would be simpler / easier to read and work just as well:

`payload := new(bytes.Buffer)` or:
```
var payload bytes.Buffer
responseSpec, err := c.httpRequest("GET", uri.String(), contentType, &payload)
```
https://golang.org/pkg/bytes/#NewBuffer - "In most cases, new(Buffer) (or just declaring a Buffer variable) is sufficient to initialize a Buffer."